File: 0974/main.cpp
class Solution {
public:
    int subarraysDivByK(vector<int>& A, int K) {
        for(int& n : A) {
            n %= K;
            if(n < 0) n += K;
        }
        vector<int> nums(K);
        nums[0] = 1;
        int count = 0, num = 0;
        for(int& n : A) {
            num += n;
            if(num >= K) num -= K;
            count += nums[num];
            ++nums[num];
        }
        return count;
    }
};

File: 0974/main.rs
impl Solution {
    pub fn subarrays_div_by_k(a: Vec<i32>, k: i32) -> i32 {
        let mut nums = vec![0; k as usize];
        nums[0] += 1;
        let (mut num, mut count): (i32, i32) = (0, 0);
        for n in a.iter() {
            num = (num + n)% k;
            if num < 0 {
                num += k;
            }
            count += nums[num as usize];
            nums[num as usize] += 1;
        }
        count
    }
}